src/wp-includes/default-filters.php

```diff
@@ -762,7 +763,9 @@
 // Update ignoredHookedBlocks postmeta for wp_navigation post type.
 add_filter( 'rest_pre_insert_wp_navigation', 'update_ignored_hooked_blocks_postmeta' );
 
-// Inject hooked blocks into the wp_navigation post type REST response.
+// Inject hooked blocks into the Posts endpoint REST response for some given post types.
```

Comment:
Do you remember why the line above was needed? ` update_ignored_hooked_blocks_postmeta` could have something with the first and last child. Maybe same handling is missing for post and pages? More broadly how the same functionality could work with CPTs?

Reply:
Good spot 👍 Yeah, `update_ignored_hooked_blocks_postmeta` is almost certainly about first/last child insertion. I need to test if that's currently working with posts and pages. If it isn't, I'll probably have to add it for those post types.

Reply:
There's also the somewhat philosophical question if a block that's hooked as `core/post-content`'s `lastChild` should be inserted if we're using a classic theme -- where there is no `core/post-content` block, strictly speaking 😅

Reply:
This seems to work (to get a block hooked has `core/post-content`'s `last_child` to be rendered on the frontend):

```diff
diff --git a/src/wp-includes/blocks/post-content.php b/src/wp-includes/blocks/post-content.php
index 25be880cc4..8be98e5926 100644
--- a/src/wp-includes/blocks/post-content.php
+++ b/src/wp-includes/blocks/post-content.php
@@ -46,6 +46,19 @@ function render_block_core_post_content( $attributes, $content, $block ) {
                $content .= wp_link_pages( array( 'echo' => 0 ) );
        }
 
+       // Wrap in Post Content block so the Block Hooks algorithm can insert blocks
+       // that are hooked as first or last child of `core/post-content`.
+       $content = get_comment_delimited_block_content(
+               'core/post-content',
+               $attributes, // TODO: Merge ignoredHookedBlocks from post meta.
+               $content
+       );
+
+       // We need to remove the `core/post-content` block wrapper after the Block Hooks algorithm,
+       // but before `do_blocks` runs, as it would otherwise attempt to render the same block again --
+       // thus recursing infinitely.
+       add_filter( 'the_content', 'remove_serialized_parent_block', 8 );
+
        /** This filter is documented in wp-includes/post-template.php */
        $content = apply_filters( 'the_content', str_replace( ']]>', ']]&gt;', $content ) );
        unset( $seen_ids[ $post_id ] );
```

This means I'll have to cover this on the GB side.

Reply:
(Updated https://github.com/WordPress/gutenberg/pull/67272 to include the required changes.)